src/openforms/formio/dynamic_config/tests/test_referentielijsten_config.py

```diff
@@ -0,0 +1,372 @@
+        self.assertEqual(
+            log.extra_data["error"],
+            _(
+                "Cannot fetch from Referentielijsten API, because no `service` is configured."
+            ),
+        )
+
+    def test_no_code_configured(self):
+        self.component["openForms"]["code"] = ""
+
+        with self.assertRaises(ServiceUnavailable):
+            Select("select").mutate_config_dynamically(
+                self.component, self.submission, {}
+            )
+
+        self.assertNotIn("data", self.component)
+        log = TimelineLogProxy.objects.get(object_id=self.submission.form.id)
+        assert log.extra_data
+        self.assertEqual(log.template, "logging/events/form_configuration_error.txt")
+        self.assertEqual(
+            log.extra_data["error"],
+            _(
+                "Cannot fetch from Referentielijsten API, because no `code` is configured."
+            ),
+        )
+
+    def test_service_does_not_exist(self):
+        self.service.delete()
+
+        with self.assertRaises(ServiceUnavailable):
+            Select("select").mutate_config_dynamically(
+                self.component, self.submission, {}
+            )
+
+        self.assertNotIn("data", self.component)
+        log = TimelineLogProxy.objects.get(object_id=self.submission.form.id)
+        assert log.extra_data
+        self.assertEqual(log.template, "logging/events/form_configuration_error.txt")
+        self.assertEqual(
+            log.extra_data["error"],
+            _(
+                "Cannot fetch from Referentielijsten API, service with {service_slug} does not exist."
+            ).format(service_slug=self.service.slug),
+        )
+
+    def test_items_not_found(self):
+        self.component["openForms"]["code"] = "non-existent"
+
+        with self.assertRaises(ServiceUnavailable):
+            Select("select").mutate_config_dynamically(
+                self.component, self.submission, {}
+            )
+
+        self.assertNotIn("data", self.component)
+        log = TimelineLogProxy.objects.get(object_id=self.submission.form.id)
+        assert log.extra_data
+        self.assertEqual(
+            log.template, "logging/events/referentielijsten_failure_response.txt"
+        )
+        self.assertEqual(
+            log.extra_data["error"],
+            _("No results found from Referentielijsten API."),
+        )
+
+    @requests_mock.Mocker()
+    def test_request_exception(self, m):
+        m.get(
+            f"{self.service.api_root}items?tabel__code=tabel1",
+            exc=RequestException("something went wrong"),
+        )
+
+        with self.assertRaises(ServiceUnavailable):
+            Select("select").mutate_config_dynamically(
+                self.component, self.submission, {}
+            )
+
+        self.assertNotIn("data", self.component)
+        log = TimelineLogProxy.objects.get(object_id=self.submission.form.id)
```

Comment:
```suggestion
        log = TimelineLogProxy.objects.for_object(self.submission).get()
```
filtering purely on ID breaks when you have log records to the same ID but a different content type - you really *must* take into account the content_type field of the GFK, and for that we have a helper.